File: SinglyLinkedList/SinglyLinkedList.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class SinglyLinkedList {
    constructor() {
        this.head = null;
    }

    insert(data) {
        const newNode = new Node(data);
        if (!this.head) {
            this.head = newNode;
        } else {
            newNode.next = this.head;
            this.head = newNode;
        }
    }

    delete(key) {
        if (!this.head) return;
        if (this.head.data === key) {
            this.head = this.head.next;
            return;
        }
        let current = this.head;
        while (current.next && current.next.data !== key) {
            current = current.next;
        }
        if (current.next) {
            current.next = current.next.next;
        }
    }

    display() {
        let current = this.head;
        let result = '';
        while (current) {
            result += current.data + ' -> ';
            current = current.next;
        }
        console.log(result + 'None');
    }
}

// Example usage
const sll = new SinglyLinkedList();
sll.insert(10);
sll.insert(20);
sll.insert(30);
sll.display();
sll.delete(20);
sll.display();

File: SinglyLinkedList/SinglyLinkedList.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class SinglyLinkedList:
    def __init__(self):
        self.head = None
    def insert(self, data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
        else:
            current = self.head
            while current.next:
                current = current.next
            current.next = new_node

    def delete(self, key):
        if not self.head:
            return
        if self.head.data == key:
            self.head = self.head.next
            return
        prev = self.head
        current = self.head.next
        while current and current.data != key:
            prev = current
            current = current.next
        if current:
            prev.next = current.next

    def display(self):
        elements = []
        current_node = self.head
        while current_node:
            elements.append(current_node.data)
            current_node = current_node.next
        print(" -> ".join(str(element) for element in elements) + " -> None")

# Example usage
sll = SinglyLinkedList()
sll.insert(10)
sll.insert(20)
sll.insert(30)
sll.display()
sll.delete(20)
sll.display()
